pkg/events/base.go

```diff
@@ -0,0 +1,52 @@
+package events
+
+import (
+	"time"
+
+	"github.com/filecoin-project/lassie/pkg/types"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multicodec"
+)
+
+type retrievalEvent struct {
+	eventTime   time.Time
+	retrievalId types.RetrievalID
+	payloadCid  cid.Cid
+}
+
+func (r retrievalEvent) Time() time.Time                { return r.eventTime }
+func (r retrievalEvent) RetrievalId() types.RetrievalID { return r.retrievalId }
+func (r retrievalEvent) PayloadCid() cid.Cid            { return r.payloadCid }
```

Comment:
Maybe we should take this opportunity to rename this to "RootCid", the term "payload" is a bit of a hangover from Filecoin storage mechanics reflected in autoretrieve, where there's a difference between the root of a stored DAG and the root of the (sub) DAG that you actually want to retrieve. I believe Estuary still maintains this difference and it appeared in their version of the indexer interface in autoretrieve.

Reply:
Does your opinion here collide with issue #248 that Magik submitted? I'm down for whatever verbage is more clear. I don't feel like I know enough to discern the differences between what the different "roots" portray to be able to make provide an opinion on this.

Reply:
mmm, there's probably a disagreement brewing around that - "root CID" is much older than filecoin; it's the root of a DAG.

"RequestCid" is an alternative; but PayloadCid is the problem one I think, aside from it being very Filecoin specific it's got a particular Filecoin meaning.

Reply:
Moved to isse #334